framework/db/transaction_manager.py

```diff
@@ -438,6 +440,44 @@ def GetByIDAsDict(self, trans_id, target_id=None):
         return self.DeriveTransactionDict(transaction_obj, include_raw_data=True)
 
     @target_required
+    def GetHrtResponse(self, filter_data, trans_id, target_id=None):
+        transaction_obj = self.db.session.query(models.Transaction).filter_by(target_id=target_id, id=trans_id).first()
+
+        if 'language' in filter_data:
+            # Validate the request. It should contain language as POST parameter.
+            language = filter_data['language']
+        else:
+            response = "Please specify the language."
+            return response
+
+        # If target not found. Raise error.
+        if not transaction_obj:
+            raise InvalidTransactionReference("No transaction with %s exists" % str(trans_id))
+
+        transaction_obj_dict = dict(transaction_obj.__dict__)
+        raw_request = transaction_obj_dict['raw_request']
+        method = transaction_obj_dict['method']
+
+        # Due to issue https://github.com/owtf/http-request-translator/issues/53
+        if method not in ['GET', 'POST']:
+            response = "Request cannnot be completed.\nOnly GET and POST method supported."
+            return response
+
+        try:
+            hrt_obj = HttpRequestTranslator(
+                request=raw_request,
+                languages=language,
+                proxy=None,
+                search_string=None,
+                data=None)
+            codes = hrt_obj.generate_code()
+            response = (''.join(v for v in codes.values()))
+        except:
```

Comment:
Please avoid this. Empty catch could introduce nasty bug that would be hard to spot, especially in a multiprocess project like OWTF.

Consider following something similar than what I did for ptp (https://github.com/owtf/owtf/blob/develop/framework/plugin/plugin_handler.py#L279). So catch the exceptions that hrt could raise and catch Exception after to log the error. It would even help to troubleshoot the error like https://github.com/owtf/ptp/issues/28